br/pkg/lightning/backend/local/local.go

```diff
@@ -1276,27 +1275,32 @@ func (local *local) writeAndIngestByRanges(ctx context.Context, engine *Engine,
 				wg.Done()
 			}()
 			var err error
-			// max retry backoff time: 2+4+8+16=30s
+			// max retry backoff time: 2+4+8+16+30*26=810s
 			backOffTime := time.Second
-			for i := 0; i < maxRetryTimes; i++ {
+		loop:
+			for i := 0; i < maxWriteAndIngestRetryTimes; i++ {
 				err = local.writeAndIngestByRange(ctx, engine, startKey, endKey, regionSplitSize, regionSplitKeys)
 				if err == nil || common.IsContextCanceledError(err) {
-					return
+					break
 				}
 				log.L().Warn("write and ingest by range failed",
 					zap.Int("retry time", i+1), log.ShortError(err))
 				backOffTime *= 2
+				if backOffTime > time.Second*30 {
+					backOffTime = time.Second * 30
+				}
 				select {
 				case <-time.After(backOffTime):
 				case <-ctx.Done():
-					return
+					err = ctx.Err()
+					break loop
```

Comment:
OK, that makes sense. Since ctx canceled error is likely to happen in every sub routine how about just check ctx.Err after the outer wg.Wait if allErr is still nil.